ios/FluentUI/AvatarGroup/AvatarGroup.swift

```diff
@@ -185,48 +186,86 @@ public struct AvatarGroup: View {
+                        let sizeDiff = !isLastDisplayed ? (currentAvatarHasRing ? avatarSizeDifference : avatarSizeDifference - ringGapOffset) :
+                            currentAvatarHasRing ? (avatarSize - ringGapOffset) - imageSize : (avatarSize - (ringGapOffset * 2)) - imageSize
+                        let x = avatarSize + tokens.interspace - ringGapOffset
+
+                        let ringPaddingInterspace = nextAvatarHasRing ? interspace - (ringOffset + ringOuterGap) : interspace - ringOffset
+                        let noRingPaddingInterspace = nextAvatarHasRing ? interspace - ringOuterGap : interspace
+                        let rtlRingPaddingInterspace = (nextAvatarHasRing ? -x - ringOuterGap : -x + ringOffset)
+                        let rtlNoRingPaddingInterspace = (nextAvatarHasRing ? -x - ringOffset - ringOuterGap : -x)
+                        let stackPadding = (currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace)
+
+                        let xPosition = currentAvatarHasRing ? x - ringOuterGap - ringOuterGap : x - ringOuterGap
+                        let xPositionRTL = currentAvatarHasRing ? rtlRingPaddingInterspace : rtlNoRingPaddingInterspace
+                        let xOrigin = Locale.current.isRightToLeftLayoutDirection() ? xPositionRTL : xPosition
+                        let yOrigin = sizeDiff / 2
+                        let cutoutSize = isLastDisplayed ? (ringOuterGap * 2) + imageSize : nextAvatarSize
+
+                        VStack {
+                            avatarView
+                                .transition(.identity)
+                                .modifyIf(needsCutout, { view in
+                                    view.mask(Avatar.AvatarCutout(
+                                                xOrigin: xOrigin,
+                                                yOrigin: yOrigin,
+                                                cutoutSize: cutoutSize)
+                                                .fill(style: FillStyle(eoFill: true)))
+                                })
+                        }
+                        .padding(.trailing, tokens.style == .stack ? stackPadding : interspace)
+                        .animation(Animation.linear(duration: 0.1))
+                        .transition(AnyTransition.move(edge: .leading))
+                    }
+                    if overflowCount > 0 {
+                        VStack {
+                            createOverflow(count: overflowCount)
+                        }
+                        .transition(AnyTransition.move(edge: .leading))
+                    }
+                }
+                .frame(height: geo.size.height,
+                       alignment: .leading)
             }
+            .frame(width: calcWidth(hasOverflow: overflowCount > 0), height: groupHeight, alignment: .leading)
+        }
+
+        return avatarGroupContent
+    }
+
+    private func calcWidth(hasOverflow: Bool) -> CGFloat {
```

Comment:
Is this considering added width of the avatars with rings?
Seems to me that the leading gap/padding is bigger when the avatar size is bigger.

![Screen Shot 2021-11-24 at 1 44 28 PM](https://user-images.githubusercontent.com/68076145/143316885-2c371684-6fbd-42ce-878a-67c407264485.png)

Reply:
And the number of avatars displayed is also influencing the calculation:
![Screen Shot 2021-11-24 at 1 54 13 PM](https://user-images.githubusercontent.com/68076145/143317512-1e747e1e-5e86-4f9a-93e5-66ad5c6c072c.png)

![Screen Shot 2021-11-24 at 1 53 54 PM](https://user-images.githubusercontent.com/68076145/143317498-7d0e0af6-458d-4c4b-b5f7-7d70bdd88b5a.png)